units/src/amount/signed.rs

```diff
@@ -480,61 +480,94 @@ impl fmt::Display for SignedAmount {
 }
 
 impl ops::Add for SignedAmount {
-    type Output = SignedAmount;
+    type Output = AmountOpResult<SignedAmount>;
 
     fn add(self, rhs: SignedAmount) -> Self::Output {
-        self.checked_add(rhs).expect("SignedAmount addition error")
+        match self.checked_add(rhs) {
```

Comment:
You could add a private `OptionExt` trait with `unwrap_or_amount_op_err` to make these simpler. :)

Reply:
You've defined it but didn't call.

Reply:
Oh I now understand this comment, I did not yesterday. Will fix.